contracts/eosio.bios/include/eosio.bios/eosio.bios.hpp

```diff
@@ -5,11 +5,37 @@
 #include <eosiolib/privileged.hpp>
 #include <eosiolib/producer_schedule.hpp>
 
+/** 
+ * EOSIO Contracts
+ * 
+ * @details The design of the EOSIO blockchain calls for a number of smart contracts that are run at a 
+ * privileged permission level in order to support functions such as block producer registration and 
+ * voting, token staking for CPU and network bandwidth, RAM purchasing, multi-sig, etc. These smart 
+ * contracts are referred to as the system, token, msig and wrap (formerly known as sudo) contracts.
+ * 
+ * This repository contains examples of these privileged contracts that are useful when deploying, 
+ * managing, and/or using an EOSIO blockchain. They are provided for reference purposes:
+ * - eosio.bios
+ * - eosio.system
+ * - eosio.msig
+ * - eosio.wrap
+ * 
+ * The following unprivileged contract(s) are also part of the system.
+ * - eosio.token
+ */
+
 namespace eosio {
+
    using eosio::permission_level;
    using eosio::public_key;
    using eosio::ignore;
 
+   /**
+    * A weighted permission. 
+    * 
+    * @details Defines a weighted permission, that is a permission which has a weight associated. 
+    * A permission is defined by an account name plus a permission name.
+    */
    struct permission_level_weight {
```

Comment:
@zorba80 I looked through these docs, and am not sure how this would be documented to provide context of the source without bloating the inline docs. Do you have something specific in mind?

Reply:
It's resolved now. That's what I had in mind, one or two sentences to provide some context.

Reply:
Thank you for helping!